src/me/mrCookieSlime/Slimefun/listeners/AncientAltarListener.java

```diff
@@ -105,16 +123,31 @@ else if (item.getID().equals("ANCIENT_ALTAR")) {
 							else {
 								altars.remove(e.getClickedBlock());
 								Messages.local.sendTranslation(e.getPlayer(), "machines.ANCIENT_ALTAR.unknown-recipe", true);
+
+								pedestals.forEach((pblock)->{
+									Variables.altarinuse.remove(pblock.getLocation());
+								});
+								Variables.altarinuse.remove(b.getLocation());  // bad recipe, no longer in use.
 							}
 						}
 						else {
 							altars.remove(e.getClickedBlock());
 							Messages.local.sendTranslation(e.getPlayer(), "machines.ANCIENT_ALTAR.unknown-catalyst", true);
+
+							pedestals.forEach((pblock)->{
+								Variables.altarinuse.remove(pblock.getLocation());
+							});
+							Variables.altarinuse.remove(b.getLocation());  // unkown catalyst, no longer in use
 						}
 					}
 					else {
 						altars.remove(e.getClickedBlock());
 						Messages.local.sendTranslation(e.getPlayer(), "machines.ANCIENT_ALTAR.not-enough-pedestals", true, new Variable("%pedestals%", String.valueOf(pedestals.size())));
+
+						pedestals.forEach((pblock)->{
+							Variables.altarinuse.remove(pblock.getLocation());
+						});
```

Comment:
I did test it... as I mentioned there seems to be some lag or maybe it's just an interruption that happens in the animation if I spam-click the pedestals/altars with ingredients during the animation... which was when the duping used to happen... now it just makes it look like I'm punching the air and the animation completely pauses until I stop... at which point the animation resumes and crafting happens normally.  I don't know if this would cause actual server lag or if it would just interrupt the player's personal position in their animation, but I've so far only tried this out on my local server which is running on a computer with 2 copies of minecraft in addition to the "server" so it's not a very good test of that.  The goal of stopping the duplication was achieved, that's the main thing... there might be a way to like.. set a time associated to the player of their last interaction and just cancel and return on any attempt of interaction if it was within so many milliseconds or whatever, but unless there's an army of players spam clicking pedestals simultaneously, i doubt that would really be that important.